pkg/download/download.go

```diff
@@ -130,6 +142,24 @@ func Download(ctx context.Context, endpoint string, opts ...Option) (b []byte, e
 func download(req *http.Request, dlOpts *downloadOptions) (data []byte, err error) {
 	client := &http.Client{}
 
+	if dlOpts.LowSrcPort {
+		port := 100 + rand.Intn(512)
+
+		localTCPAddr, tcperr := net.ResolveTCPAddr("tcp", ":"+strconv.Itoa(port))
```

Comment:
I don't see though how this would handle already used ports?

Reply:
No. I do not. If port is busy it returns error, and Download() retry the request.